src/pybind/mgr/dashboard/services/service.py

```diff
@@ -221,3 +221,47 @@ def unset_rgw_hostname(self, daemon_name: str):
         rgw_hostname_setting = Settings.RGW_HOSTNAME_PER_DAEMON
         rgw_hostname_setting.pop(daemon_name, None)
         Settings.RGW_HOSTNAME_PER_DAEMON = rgw_hostname_setting
+
+    def get_username_from_realm_name(self, realm_name: str) -> str:
```

Comment:
The username seems to be really dug out which makes me think if we need to do this ?
Can it be just passed via UI as currently ?

Reply:
I agree its a bit dugout but for the ease of user, its a necessary step. On the multi-site page, if we edit a zone, we can see the access/secret keys of the user. So even if we want to get the username as input from the user, the user needs to 
1. Go the Users page
2. Match access/secret keys of the zone by expanding each user row and check to which user these keys belong to and then they'l be able to get the username.
IMO we should avoid this and include this in automation

Reply:
Can't we provide a selection box to user there ? and fetch the secret/access keys of that user ?
I guess that would be eaiser and user friendly as well.

The current approach does not seem the best approach IMO

Reply:
The thing is it will be still difficult for the user to find the username associated with the realm. The user still needs to find the username by going through the steps I mentioned above and then select that user from the selection box. IMO automation means minimal or no user intervention even if it gets technically complex in the backend, So I would still go with the current approach.